File: src/test/java/com/snackgame/server/fixture/BestScoreFixture.java
package com.snackgame.server.fixture;

import static com.snackgame.server.fixture.SeasonFixture.베타시즌;
import static com.snackgame.server.fixture.SeasonFixture.시즌1;
import static com.snackgame.server.game.metadata.Metadata.APPLE_GAME;
import static com.snackgame.server.game.metadata.Metadata.SNACK_GAME;
import static com.snackgame.server.member.fixture.MemberFixture.땡칠;
import static com.snackgame.server.member.fixture.MemberFixture.똥수;
import static com.snackgame.server.member.fixture.MemberFixture.유진;
import static com.snackgame.server.member.fixture.MemberFixture.정언;
import static com.snackgame.server.member.fixture.MemberFixture.정환;

import com.snackgame.server.member.fixture.MemberFixture;
import com.snackgame.server.rank.domain.BestScore;
import com.snackgame.server.support.fixture.FixtureSaver;

@SuppressWarnings("NonAsciiCharacters")
public class BestScoreFixture {

    public static BestScore 사과게임_베타시즌_똥수_10점() {
        return new BestScore(똥수().getId(), APPLE_GAME.getGameId(), 베타시즌().getId(), 1, 10, 1);
    }

    public static BestScore 사과게임_베타시즌_땡칠_10점() {
        return new BestScore(땡칠().getId(), APPLE_GAME.getGameId(), 베타시즌().getId(), 2, 10, 2);
    }

    public static BestScore 사과게임_베타시즌_정환_8점() {
        return new BestScore(정환().getId(), APPLE_GAME.getGameId(), 베타시즌().getId(), 3, 8, 3);
    }

    public static BestScore 사과게임_베타시즌_유진_6점() {
        return new BestScore(유진().getId(), APPLE_GAME.getGameId(), 베타시즌().getId(), 4, 6, 4);
    }

    public static BestScore 사과게임_베타시즌_정언_8점() {
        return new BestScore(정언().getId(), APPLE_GAME.getGameId(), 베타시즌().getId(), 5, 8, 5);
    }

    public static BestScore 사과게임_시즌1_땡칠_20점() {
        return new BestScore(땡칠().getId(), APPLE_GAME.getGameId(), 시즌1().getId(), 6, 20, 6);
    }

    public static BestScore 사과게임_시즌1_정환_20점() {
        return new BestScore(정환().getId(), APPLE_GAME.getGameId(), 시즌1().getId(), 7, 20, 7);
    }

    public static BestScore 사과게임_시즌1_유진_20점() {
        return new BestScore(유진().getId(), APPLE_GAME.getGameId(), 시즌1().getId(), 8, 20, 8);
    }

    public static BestScore 사과게임_시즌1_정언_8점() {
        return new BestScore(정언().getId(), APPLE_GAME.getGameId(), 시즌1().getId(), 9, 8, 9);
    }

    public static BestScore 스낵게임_시즌1_땡칠_20점() {
        return new BestScore(땡칠().getId(), SNACK_GAME.getGameId(), 시즌1().getId(), 11, 20, 10);
    }

    public static BestScore 스낵게임_시즌1_유진_20점() {
        return new BestScore(유진().getId(), SNACK_GAME.getGameId(), 시즌1().getId(), 10, 20, 11);
    }

    public static BestScore 스낵게임_시즌1_정언_8점() {
        return new BestScore(정언().getId(), SNACK_GAME.getGameId(), 시즌1().getId(), 13, 8, 13);
    }

    public static void saveAll() {
        MemberFixture.saveAll();
        SeasonFixture.saveAll();
        FixtureSaver.save(
                사과게임_베타시즌_똥수_10점(),
                사과게임_베타시즌_땡칠_10점(),
                사과게임_베타시즌_정환_8점(),
                사과게임_베타시즌_유진_6점(),
                사과게임_베타시즌_정언_8점(),
                사과게임_시즌1_땡칠_20점(),
                사과게임_시즌1_정환_20점(),
                사과게임_시즌1_유진_20점(),
                사과게임_시즌1_정언_8점(),
                스낵게임_시즌1_땡칠_20점(),
                스낵게임_시즌1_유진_20점(),
                스낵게임_시즌1_정언_8점()
        );
    }
}

File: src/test/java/com/snackgame/server/fixture/BestScoreFixture.kt
@file:Suppress("NonAsciiCharacters")

package com.snackgame.server.fixture

import com.snackgame.server.game.metadata.Metadata
import com.snackgame.server.member.fixture.MemberFixture
import com.snackgame.server.member.fixture.MemberFixture.땡칠
import com.snackgame.server.member.fixture.MemberFixture.똥수
import com.snackgame.server.member.fixture.MemberFixture.유진
import com.snackgame.server.member.fixture.MemberFixture.정언
import com.snackgame.server.member.fixture.MemberFixture.정환
import com.snackgame.server.rank.domain.BestScore
import com.snackgame.server.support.fixture.FixtureSaver

object BestScoreFixture {
    @JvmStatic
    fun 사과게임_베타시즌_똥수_10점(): BestScore {
        return BestScore(똥수().id, Metadata.APPLE_GAME.gameId, SeasonFixture.베타시즌().id, 1, 10, id = 1)
    }

    @JvmStatic
    fun 사과게임_베타시즌_땡칠_10점(): BestScore {
        return BestScore(땡칠().id, Metadata.APPLE_GAME.gameId, SeasonFixture.베타시즌().id, 2, 10, id = 2)
    }

    @JvmStatic
    fun 사과게임_베타시즌_정환_8점(): BestScore {
        return BestScore(정환().id, Metadata.APPLE_GAME.gameId, SeasonFixture.베타시즌().id, 3, 8, id = 3)
    }

    @JvmStatic
    fun 사과게임_베타시즌_유진_6점(): BestScore {
        return BestScore(유진().id, Metadata.APPLE_GAME.gameId, SeasonFixture.베타시즌().id, 4, 6, id = 4)
    }

    @JvmStatic
    fun 사과게임_베타시즌_정언_8점(): BestScore {
        return BestScore(정언().id, Metadata.APPLE_GAME.gameId, SeasonFixture.베타시즌().id, 5, 8, id = 5)
    }

    @JvmStatic
    fun 사과게임_시즌1_땡칠_20점(): BestScore {
        return BestScore(땡칠().id, Metadata.APPLE_GAME.gameId, SeasonFixture.시즌1().id, 6, 20, id = 6)
    }

    @JvmStatic
    fun 사과게임_시즌1_정환_20점(): BestScore {
        return BestScore(정환().id, Metadata.APPLE_GAME.gameId, SeasonFixture.시즌1().id, 7, 20, id = 7)
    }

    @JvmStatic
    fun 사과게임_시즌1_유진_20점(): BestScore {
        return BestScore(유진().id, Metadata.APPLE_GAME.gameId, SeasonFixture.시즌1().id, 8, 20, id = 8)
    }

    @JvmStatic
    fun 사과게임_시즌1_정언_8점(): BestScore {
        return BestScore(정언().id, Metadata.APPLE_GAME.gameId, SeasonFixture.시즌1().id, 9, 8, id = 9)
    }

    @JvmStatic
    fun 스낵게임_시즌1_땡칠_20점(): BestScore {
        return BestScore(땡칠().id, Metadata.SNACK_GAME.gameId, SeasonFixture.시즌1().id, 11, 20, id = 10)
    }

    @JvmStatic
    fun 스낵게임_시즌1_유진_20점(): BestScore {
        return BestScore(유진().id, Metadata.SNACK_GAME.gameId, SeasonFixture.시즌1().id, 10, 20, id = 11)
    }

    @JvmStatic
    fun 스낵게임_시즌1_정언_8점(): BestScore {
        return BestScore(정언().id, Metadata.SNACK_GAME.gameId, SeasonFixture.시즌1().id, 13, 8, id = 13)
    }

    @JvmStatic
    fun saveAll() {
        MemberFixture.saveAll()
        SeasonFixture.saveAll()
        FixtureSaver.save(
            사과게임_베타시즌_똥수_10점(),
            사과게임_베타시즌_땡칠_10점(),
            사과게임_베타시즌_정환_8점(),
            사과게임_베타시즌_유진_6점(),
            사과게임_베타시즌_정언_8점(),
            사과게임_시즌1_땡칠_20점(),
            사과게임_시즌1_정환_20점(),
            사과게임_시즌1_유진_20점(),
            사과게임_시즌1_정언_8점(),
            스낵게임_시즌1_땡칠_20점(),
            스낵게임_시즌1_유진_20점(),
            스낵게임_시즌1_정언_8점()
        )
    }
}
